frontend/awx/interfaces/Inventory.ts

```diff
@@ -1,6 +1,78 @@
 import { Inventory as SwaggerInventory } from './generated-from-swagger/api';
+import {
+  SummaryFieldsByUser,
+  SummaryFieldsOrganization,
+  SummeryFieldObjectRole,
+} from './summary-fields/summary-fields';
+import { Label } from './Label';
 
-export interface Inventory extends Omit<SwaggerInventory, 'id' | 'name' | 'summary_fields'> {
+export interface CommonInventory
+  extends Omit<
+    SwaggerInventory,
+    'id' | 'name' | 'type' | 'kind' | 'summary_fields' | 'organization'
+  > {
   id: number;
   name: string;
+  type: 'inventory';
+  summary_fields: {
+    organization: SummaryFieldsOrganization;
+    created_by: SummaryFieldsByUser;
+    modified_by: SummaryFieldsByUser;
+    object_roles: {
+      admin_role: SummeryFieldObjectRole;
+      update_role: SummeryFieldObjectRole;
+      adhoc_role: SummeryFieldObjectRole;
+      use_role: SummeryFieldObjectRole;
+      read_role: SummeryFieldObjectRole;
+    };
+    user_capabilities: {
+      edit: boolean;
+      delete: boolean;
+      copy: boolean;
+      adhoc: boolean;
+    };
+  };
+  organization: number;
 }
+
+export interface ConstructedInventory extends CommonInventory {
+  kind: 'constructed';
+  source_vars: string;
+  update_cache_timeout: number;
+  limit: string;
+  verbosity: number;
+  summary_fields: {
+    organization: SummaryFieldsOrganization;
+    created_by: SummaryFieldsByUser;
+    modified_by: SummaryFieldsByUser;
+    object_roles: {
+      admin_role: SummeryFieldObjectRole;
+      update_role: SummeryFieldObjectRole;
+      adhoc_role: SummeryFieldObjectRole;
+      use_role: SummeryFieldObjectRole;
+      read_role: SummeryFieldObjectRole;
+    };
+    user_capabilities: {
+      edit: boolean;
+      delete: boolean;
+      copy: boolean;
+      adhoc: boolean;
+    };
+    labels: {
+      count: number;
+      results: Label[];
+    };
+  };
+}
+
+export interface RegularInventory extends CommonInventory {
+  kind: '';
+  verbosity?: never;
```

Comment:
@jamestalton I wrote a hook that converts the value that the api gives us for verbosity into the string that we want to show in the UI: frontend/awx/common/useVerbosityString.tsx.  In the case of inventories, verbosity is only available for constructed inventories and I couldn't figure out how to call that hook without the linter getting mad that verbosity may not be defined.  To get around that I went down the `never` route but I'm not sure how accepted this route is.  You can reproduce the scenario I'm talking about by removing this line and then going and looking at my call to useVerbosityString in InventoryDetails.